rmgpy/molecule/filtration.py

```diff
@@ -0,0 +1,331 @@
+                filtered_list += unique_charged_list
+
+        else:
+            # There are structures with different charge spans, but the species is not a radical.
+            # Additional charge span levels are removed
+            filtered_list = [filtered for index, filtered in enumerate(filtered_list) if
+                             charge_span_list[index] == min_charge_span]
+
+    return filtered_list
+
+
+def mark_unreactive_structures(filtered_list, mol_list):
+    """
+    Mark selected structures in filtered_list with the Molecule.reactive flag set to `False` (it is `True` by default)
+    Changes the filtered_list object, and does not return anything
+    """
+
+    # sort all structures in filtered_list so that the reactive ones are first
+    filtered_list.sort(key=lambda mol: mol.reactive, reverse=True)
+
+    # Make sure that the (first) original structure is always first in the list (unless it was filtered out).
+    # Important whenever Species.molecule[0] is expected to be used (e.g., training reactions) after generating
+    # resonance structures. However, if it was filtered out, it should be appended to the end of the list.
+    for index, filtered in enumerate(filtered_list):
+        if filtered.copy(deep=True).isIsomorphic(mol_list[0].copy(deep=True)):
+            filtered_list.insert(0, filtered_list.pop(index))
+            break
+    else:
+        # Append the original structure to list and set `reactive` to `False`.
+        # This structure may very well deviate from the octet rule or have other attributes by which it should have
+        # been filtered out. However, for processing reactions (e.g., degeneracy calculations) it should be kept
+        # (e.g., [::N]O <=> [::N][::O.] + [H.], where [::N][::O.] should be recognized as [:N.]=[::O]).
+        mol = mol_list[0]
+        logging.debug("Setting the unrepresentative resonance structure {0} as unreactive in species {1}.".format(
+            mol.toSMILES(),filtered_list[0].toSMILES()))
+        logging.debug("Unreactive structure:\n{0}\nA representative reactive structure in this species:\n{1}\n".format(
+            mol.toAdjacencyList(),filtered_list[0].toAdjacencyList()))
+        mol.reactive = False
+        filtered_list.append(mol)
+
+
+def check_reactive(filtered_list):
+    """
+    Check that there's at least one reactive structure in the returned list.
+    If not, raise an error (does not return anything)
+    """
+
+    if not any([mol.reactive for mol in filtered_list]):
+        logging.info('\n\n')
+        logging.error('No reactive structures were attributed to species {0}'.format(filtered_list[0].toSMILES()))
+        for mol in filtered_list:
+            logging.info('Structure: {0}\n{1}Reactive: {2}'.format(mol.toSMILES(),mol.toAdjacencyList(),mol.reactive))
+        logging.info('\n')
+        raise AssertionError('Each species must have at least one reactive structure. Something probably went wrong'
```

Comment:
I still think it would be a good idea to create a ResonanceError class to be used here.